src/plugins/lightscript.js

```diff
@@ -65,64 +65,44 @@ pp.rewriteOperator = function (node) {
   }
 };
 
-pp.parseForFrom = function (node, iterator) {
-  this.expectContextual("from");
-  // `for i from`
-  const arrayOrRangeStart = this.parseExpression(true);
-  if (this.match(tt._thru) || this.match(tt._til)) {
-    return this.parseForFromRange(node, iterator, arrayOrRangeStart);
-  } else {
-    return this.parseForFromArray(node, iterator, null, arrayOrRangeStart);
-  }
+const MATCHING_ITER_VARS = {
+  "idx": "elem",
+  "elem": "idx",
+  "key": "val",
+  "val": "key",
 };
 
-pp.parseForFromArray = function (node, iterator, element = null, array = null) {
-  if (iterator.type !== "Identifier") this.unexpected(iterator.start);
-  if (element && element.type !== "Identifier") this.unexpected(element.start);
-  node.id = iterator;
-  node.elem = element;
-
-  if (array) {
-    node.array = array;
-  } else {
-    this.expectContextual("from");
-    node.array = this.parseMaybeAssign(true);
+pp.parseForInIterationVariable = function (node, targetType = null) {
+  const iterType = this.state.value;
+  if (targetType && targetType !== iterType) {
+    this.unexpected(null, `Unexpected token, expected ${targetType}`);
```

Comment:
Yes, surprising though it may appear, this is correct. I had a lot of trouble with this last night, and I ran it through a debugger.

`ParserPrototype.unexpected()` expects a `TokenType` as its second argument. If you just pass it a string, it doesn't give a sensible message. There is a test case here: `lightscript/for-in/mismatched-types`